### PR TITLE
rework center to GPS functionality CU-861me5mj1

### PR DESCRIPTION
**Pull request by alexbruy**
- do not center to GPS position when switching between view and digitizing
 - add follow GPS mode when map follows GPS position

Viewing
Peek 2023-03-16 12-28

Digitizing
Peek 2023-03-16 12-28

#861me5mj1

### CODE REVIEW

**Review by tomasMizera — CHANGES_REQUESTED** [review]

File: app/qml/map/MapCanvas.qml [hunk 1]
```diff
@@ -184,17 +182,20 @@ Item {
...
       root.longPressed(point.position)
       longPressActive = true
+      userInteractedWithMap()
     }
 
     onPressedChanged: {
       if (longPressActive)
         root.longPressReleased()
       longPressActive = false
+      userInteractedWithMap()
```
[the remark this answers is not in the document]
Comment (alexbruy) [reply app/qml/map/MapCanvas.qml]: If you say so

File: app/qml/main.qml [hunk 2]
```diff
@@ -270,16 +270,16 @@ ApplicationWindow {
           mapThemesPanel.visible = true
           stateManager.state = "misc"
         }
-        onMyLocationClicked: map.centerToPosition()
 
-        onMyLocationHold: {
-            __appSettings.autoCenterMapChecked = !__appSettings.autoCenterMapChecked
-            showMessage( __appSettings.autoCenterMapChecked ?  qsTr("GPS auto-center mode on") : qsTr("GPS auto-center mode off") )
+        onMyLocationClicked: {
```
Comment (tomasMizera) [inline app/qml/main.qml:274]: @alexbruy for the sake of keeping the UX same as before, can you also add a slot for `onMyLocationHold` that would do exactly the same as `onMyLocationClicked`? Thanks!

File: app/qml/map/RecordingTools.qml [hunk 3]
```diff
@@ -182,29 +181,33 @@ Item {
...
       {
         mapTool.recordingType = RecordingMapTool.StreamMode
 
         // add first point immediately
         mapTool.addPoint( crosshair.recordPoint )
-        root.map.mapSettings.setCenter( mapPositioning.mapPosition )
-      }
-      else
-      {
-        mapTool.recordingType = RecordingMapTool.Manual
```
Comment (tomasMizera) [inline app/qml/map/RecordingTools.qml:207]: Let's keep the `else` branch here so that it is possible to exit the streaming mode

**Review by tomasMizera — APPROVED** [review]
Let's proceed with testing :)

**Comment by tomasMizera** [comment]
#861me5mj1

**Comment by PeterPetrik** [comment]
@alexbruy can you please check to merge to dev-2.3 and rebase? thanks

**Comment by tomasMizera** [comment]
This is stale for too long now, I will cherry-pick some changes and create a new PR

**Pull request closed without merging**